### ISSUE TITLE
improve backend error handling

### ISSUE THREAD

**Issue by lukavdplas** [UUDigitalHumanitieslab/I-analyzer#906]
For visualisations, downloads and the like, errors can be handled more consistently:

If errors are properly caught, views are currently about 50/50 in what kind of response they give. Some views return an error status, e.g.

```python
abort(500, 'something went wrong')
```

(or more often than not, without the message), which can be handled in the frontend with

```typescript
getResults().then(this.doSomething).catch(this.showError);
```

Other views return a status 200 response like

```python
return jsonify({'success': False, 'message': 'something went wrong'})
```

which can be handled as

```typescript
getResults().then(response => {
	if (response.success) {
		this.doSomething(response.data);
	} else {
		this.showError(response.message)
	}
});
```

Sometimes these two methods are even mixed within a single view. Some frontend components do not catch errors properly as they only expect one of these as potential responses. It's also arbitrary which one is implemented for a view.

I'm not sure why we use the latter method at all, is there a reason to prefer this? 

Another point of improvement: errors should be logged. We make some log statements during indexing, but the visualisation and download views barely use the logger.
